### PR TITLE
BatchBalancer & BatchDiscount Post-HyperDrive Adjustment

### PR DESCRIPTION
**Pull request by zixuanzh**
Discussion issue #173

### CODE REVIEW

**Review by arajasek — COMMENTED** [review]
I believe `PreCommitBatch` currently allows for a batch size of one. What impact would the new BatchBalancer / BatchDiscount have on single-sector "batches"?

File: FIPS/fip-balancer_post_hyperdrive.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+```
+CrossoverNetworkBaseFee = BatchBalancer * BatchDiscount * SingleProofGasUsage / (SingleProofGasUsage - BatchProofGasUsage / NumProofsBatched)
+```
+
+This equilibrium crossover BaseFee in turn impacts other metrics such as the cost of PublishStorageDeals, protocol revenue for a particular growth rate, and the return of storage provision. You can find the table below for a comparison between existing and proposed Balancer values. Note that the estimate can be off and network participants can choose to deviate from the equilibrium.
+
+| BatchBalancer | Estimated Crossover Network BaseFee | Estimated PublishStorageDeals Network Fee | Estimated 32 GiB Sector Network Fee | Estimated 32 GiB Sector Annual Return | Estimated Daily Protocol Revenue at Current Growth Rate |
```
Comment (jennijuju) [inline FIPS/fip-balancer_post_hyperdrive.md:84]: Can we please  specify  how PSD network fee is estimated? More specifically, how many deals in a single message?

File: FIPS/fip-balancer_post_hyperdrive.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+| ----------- | ----------- | ----------- | ----------- | ----------- | ----------- |
+| 2 nanoFIL   | ~0.12 nanoFIL       | ~0.0064 FIL       | ~0.0085 FIL       | ~0.165 FIL       | ~13k FIL       |
+| 5 nanoFIL   | ~0.32 nanoFIL        | ~0.016 FIL       | ~0.021 FIL       | ~0.165 FIL         | ~34k FIL       |
+| 7 nanoFIL   | ~0.45 nanoFIL        | ~0.023 FIL       | ~0.030 FIL       | ~0.165 FIL       | ~47k FIL       |
+| 10 nanoFIL   | ~0.64 nanoFIL        | ~0.032 FIL       | ~0.042 FIL       | ~0.165 FIL       | ~68k FIL       |
+
+Note that Network Fee is halved for 64 GiB sectors and the above unit cost numbers go down further as the number of aggregation increases. This FIP currently proposes 5 nanoFIL for BatchBalancer. However, if the community, especially storage providers, believe that 7 nanoFIL is a better value, we would support that change too.
```
Comment (jennijuju) [inline FIPS/fip-balancer_post_hyperdrive.md:91]: Please correct me if I’m wrong - but I don’t thinks PSD network fee will be halved for 64 gib sectors, it should be irrelevant?

File: FIPS/fip-balancer_post_hyperdrive.md (hunk 1 above)
Comment (jennijuju) [reply FIPS/fip-balancer_post_hyperdrive.md:84]: What does 32 gib sector network fee include? Precommit + provecommit gas cost?

File: FIPS/fip-balancer_post_hyperdrive.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+## Product Considerations
+All FIPs should optimize for long-term network health. This FIP balances many difficult tradeoffs to achieve that goal - aiming to align all network participants with the network’s long term health and growth. Recalibrating network parameters to observed network growth rates as this FIP proposes is critical to that aim. However, one negative side-effect is a roughly-estimated ~.009 FIL increase in the gas costs of PublishStorageDeal messages, which has negative product considerations for storage providers accepting storage deals / participating in the deal-making market. However, the significant overall demand and return for storage providers accepting FIL+ deals after this increase is still strongly rational for storage providers as amortized over the length of the deal.
```
Comment (jennijuju) [inline FIPS/fip-balancer_post_hyperdrive.md:94]:
```suggestion
All FIPs should optimize for long-term network health. This FIP balances many difficult tradeoffs to achieve that goal - aiming to align all network participants with the network’s long term health and growth. Recalibrating network parameters to observed network growth rates as this FIP proposes is critical to that aim. However, one negative side-effect is a roughly-estimated ~.009 FIL increase in the gas costs of `PublishStorageDeals` messages, which has negative product considerations for storage providers accepting storage deals / participating in the deal-making market. However, the significant overall demand and return for storage providers accepting FIL+ deals after this increase is still strongly rational for storage providers as amortized over the length of the deal.
```

File: FIPS/fip-balancer_post_hyperdrive.md (hunk 1 above)
Comment (jennijuju) [reply FIPS/fip-balancer_post_hyperdrive.md:84]: moved to the issue

File: FIPS/fip-balancer_post_hyperdrive.md (hunk 3 above)
Comment (zixuanzh) [reply FIPS/fip-balancer_post_hyperdrive.md]: resolved in new commit

**Review by momack2 — APPROVED** [review]

**Comment by AliceFil** [comment]
This FIP is good for the whole network. As a member of community, I would like to support it.

**Pull request merged**